### PR TITLE
Importing components with @using directives

### LINKED ISSUE #11829: [Preview 4] Blazor @using directive support
More info here https://github.com/aspnet/AspNetCore/issues/5577.

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #11829 

Moved from https://github.com/aspnet/AspNetCore.Docs/pull/11961.

I reacted to the feedback ... it may still need a spot of work here and there.

### CODE REVIEW

File: aspnetcore/blazor/components.md [hunk 1]
```diff
@@ -521,6 +521,44 @@ The [sample app](https://github.com/aspnet/Docs/tree/master/aspnetcore/blazor/co
 
 The base class should derive from `ComponentBase`.
 
+## Import components
+
+The namespace of a component authored with Razor is based on:
+
+* The project's `RootNamespace`.
+* The path from the project root to the component. For example, `ComponentsSample/Pages/Index.razor` is in the namespace `ComponentsSample.Pages`. All components in the same folder, `Pages`, and the parent folder, `ComponentsSample`, are in scope.
```
Comment (ajaybhargavb) [inline aspnetcore/blazor/components.md]: I think the third sentence here can be modified to something like 
"Components follow c# name binding rules. So in the case of `Index.razor`, all components in the same folder, `Pages`, and the parent folder, `ComponentsSample`, are in scope."

File: aspnetcore/blazor/components.md [hunk 2]
```diff
@@ -521,6 +521,44 @@ The [sample app](https://github.com/aspnet/Docs/tree/master/aspnetcore/blazor/co
...
+> [!NOTE]
+> The `global::` qualification isn't supported.
+>
+> Importing components with aliased usings (for example, `@using Foo = Bar`) isn't supported.
+>
+> Partially qualified names aren't supported. For example, adding `@using ComponentsSample` and referencing `NavMenu.razor` with `<Shared.NavMenu></Shared.NavMenu>` isn't supported.
+
+## Importing components
+
+Components infer their namespace based on the file path and the file name.
```
Comment (ajaybhargavb) [inline aspnetcore/blazor/components.md]: This line seems to be misplaced.

**Comment by guardrex** [comment]
Note that an invalid file link popped up. It wasn't caused by this PR:

aspnetcore/signalr/javascript-client.md

Invalid file link:(~/signalr/javascript/api/@aspnet/signalr/hubconnectionstate)

I'll investigate and patch it on another PR in a few minutes.

**Comment by isc30** [comment]
Hi, seems like importing components from external libraries stops working after this change

**Comment by guardrex** [comment]
@isc30 Open an engineering issue for it if you're having trouble on the aspnet/AspNetCore issues: https://github.com/aspnet/AspNetCore/issues

Note that for doc issues, please use the **This Page** feedback button at the bottom of the topic. It tags the issue with metadata for tracking. Also, we don't work closed issues and merged PRs. Best to just open a new issue for any trouble that you run into.

**Comment by isc30** [comment]
oh sorry, didn't notice that this was in a different repo. thanks

**Pull request merged**